src/main.c

```diff
@@ -1093,6 +1093,7 @@ void restart_daemon(void *d1, void *d2, void *d3, void *d4) {
   }
 
   pr_event_generate("core.postparse", NULL);
+  pr_event_generate("core.restarted", NULL);
```

Comment:
This new `core.restarted` event is the main reason why this module requires ProFTPD 1.3.10rc1, and not earlier versions (as they did not generate this event).  This event is necessary for notifying systemd of when ProFTPD is reloading, per the requirements of the `Type=notify-reload` directive in the unit file.